### PR TITLE
address PR #51 feedback and align Linux-first guidance

### PR DESCRIPTION
**Pull request by 3clyp50**
This commit closes the actionable PR #51 review feedback and then updates the active project guidance so the next tranche starts from the real current baseline instead of stale migration-era assumptions.

The most important part of this work is the PR #51 polish and hardening pass:
- viewer URL and launch-token secrecy fixes
- async viewer bootstrap reliability improvements
- robust Orthanc credential templating
- de-identified clinical seed logging

It also updates the maintained guidance to reflect:
- the current post-Phase-4-polish RadSysX baseline
- Linux-first development/validation posture
- project operating rules to reduce future drift

#### Viewer bootstrap hardening

The clinical viewer bootstrap was tightened to eliminate sensitive URL leakage and improve reliability.

Changes:
- replaced synchronous XHR with async `fetch`
- added timeout-bounded bootstrap requests
- stopped writing `StudyInstanceUIDs` / `SeriesInstanceUIDs` back into viewer URLs
- stopped leaking the opaque `launch` token through `/login?next=...`
- moved launch-token handoff for unauthenticated redirects into `sessionStorage`
- cleared persisted launch state after successful resolve and on failed resolve paths

Files:
- `viewer/assets/radsysx-bootstrap.js`

#### OHIF runtime integration cleanup

The viewer no longer depends on URL query parameters for governed study context.

Changes:
- added a RadSysX-owned OHIF dicomweb data source wrapper
- let OHIF read governed launch context from resolved RadSysX state instead of URL params
- preserved series filtering by injecting launch-derived series filters inside the data source layer
- kept the bootstrap layer focused on session validation, launch resolution, and startup cleanup

Files:
- `viewer/assets/radsysx-ohif-extension.js`
- `viewer/scripts/build-ohif-dist.mjs`

#### Orthanc config templating fix

The compose-based Orthanc startup path was hardened so credentials are expanded correctly and safely.

Changes:
- replaced the fragile inline `sed` templating command in `docker-compose.yml`
- added a dedicated `render-orthanc-config.sh` helper
- handled characters that previously broke config rendering or JSON output, including `/`, `&`, `|`, `\`, and `"`
- removed dependence on shell-quoting behavior that could leave literal env variable strings in Orthanc config

Files:
- `deploy/clinical-stack/render-orthanc-config.sh`
- `docker-compose.yml`

#### Logging hygiene

The clinical Orthanc seeding path no longer logs accession numbers.

Changes:
- replaced identifier-bearing seed logs with generic sample ordinals plus modality
- extended tests so seeded study UIDs, patient names, and accession numbers are all excluded from log output

Files:
- `backend/clinical/seed_orthanc.py`
- `backend/tests/test_clinical_platform.py`

### 2. Refreshed guidance for the current baseline

The maintained guidance set was updated to describe the actual current state of the repo, not earlier migration checkpoints.

Updated:
- `AGENTS.md`
- `README.md`
- `CLAUDE.md`
- `WARP.md`
- `PHASE4_CLINICAL_EXECUTION_CHECKLIST.md`

Changes:
- removed stale “after Phase 3” / old-checklist framing
- reframed the checklist around the current post-Phase-4-polish baseline
- removed hardcoded branch-era references that no longer belong in standing guidance
- made the environment posture Linux-first and Linux-generic
- removed WSL / Docker Desktop / Windows-specific guidance assumptions
- documented that new-machine work should begin with quick recon, then wait for the first user-reported Linux runtime test result before widening the change scope

### 3. Added durable project operating rules

`AGENTS.md` now includes a concise operating rule set based on issues encountered during this tranche.

New rule themes:
- update docs in the same tranche as behavior/env/architecture changes
- keep `AGENTS.md` authoritative and companion docs aligned
- preserve the research-vs-clinical boundary
- prefer backend-authoritative contracts for governed workflows
- keep the viewer bootstrap minimal
- avoid normalizing insecure or machine-local hacks into maintained guidance
- make environment setup reproducible from the repo
- turn recurring review findings into durable guidance
- clearly state what was and was not verified

## Why This Matters

PR #51 surfaced the right class of issues for a governed clinical runtime:
- launch/session secrecy regressions
- brittle bootstrap behavior
- incorrect compose templating under real credential values
- identifier-bearing logs
- guidance drift away from the shipped runtime

This commit fixes those issues and leaves the repo in a cleaner state for the next Linux-based validation and the next clinical tranche.

## Validation

Passed:
- `python3 -m compileall backend/clinical backend/server.py`
- `PYTHONPATH=/tmp/novion-phase3-pydeps-fast python3 -m pytest backend/tests/test_clinical_platform.py --capture=no`
  - result: `16 passed, 1 skipped`
- Orthanc templating edge-case validation with special-character credentials
- `npm run type-check --workspace viewer`
- `npm run build --workspace viewer`

Notes:
- the viewer checks passed successfully
- the final guidance-only edits were not followed by another runtime test pass because they did not change executable behavior

## Follow-Up

The next tranche should start from the updated guidance and current runtime baseline:

1. do quick recon on the Linux machine
2. wait for the first user-reported Linux runtime test results
3. triage observed issues first
4. then continue with deeper OHIF-native integration and governed SR/SEG export-reload work

## Summary by CodeRabbit

* **Documentation**
  * Added clearer Linux-focused setup and validation procedures with explicit installation steps.
  * Updated host environment prerequisites and bootstrap guidance.

* **Security & Features**
  * Improved launch token handling with secure URL parameter stripping.
  * Implemented environment variable-based credential configuration.
  * Enhanced DICOM viewer launch context filtering.

* **Bug Fixes**
  * Strengthened asynchronous bootstrap flow for improved reliability.

### CODE REVIEW

**Comment by 3clyp50** [comment]
@coderabbitai full review

File: viewer/scripts/build-ohif-dist.mjs [hunk 1]
```diff
@@ -76,14 +74,14 @@ function writeAppConfig() {
     defaultDataSourceName: "dicomweb",
     dataSources: [
       {
-        namespace: "@ohif/extension-default.dataSourcesModule.dicomweb",
+        namespace: "@radsysx/extension-clinical.dataSourcesModule.dicomweb",
```
[the remark this answers is not in the document]
Comment (3clyp50) [reply viewer/scripts/build-ohif-dist.mjs:77]: Verified against OHIF's own config/runtime registration. Data source modules are namespaced as `<extensionId>.dataSourcesModule.<name>`, so the `dataSourcesModule` segment is required here. I kept that namespace and addressed the related deployment issue by removing baked `/viewer` and `/dicom-web` paths from the generated OHIF config.

File: deploy/clinical-stack/render-orthanc-config.sh [hunk 2]
```diff
@@ -0,0 +1,24 @@
+#!/bin/sh
+
+set -eu
```
[the remark this answers is not in the document]
Comment (3clyp50) [reply deploy/clinical-stack/render-orthanc-config.sh:3]: Not applying this one. The helper is intentionally `#!/bin/sh` because the Orthanc image only guarantees POSIX `sh`; `pipefail` is not portable there and would make the script less reliable. I added an inline comment documenting that constraint.

File: deploy/clinical-stack/render-orthanc-config.sh [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+escape_json_sed() {
+  printf '%s' "$1" | sed \
+    -e 's/\\/\\\\/g' \
+    -e 's/"/\\"/g' | sed \
+    -e 's/[&|\\]/\\&/g'
+}
```
[the remark this answers is not in the document]
Comment (3clyp50) [reply deploy/clinical-stack/render-orthanc-config.sh:12]: Verified this with sample credentials containing backslashes, quotes, `&`, and `|`. The suggested single-pass simplification breaks JSON or replacement rendering for those cases, so I kept the two-stage escape: first for JSON string encoding, then for the sed replacement payload. I added a comment to document why.

File: viewer/assets/radsysx-ohif-extension.js [hunk 4]
```diff
@@ -370,6 +370,57 @@
...
+          createDataSource(configuration) {
+            const defaultExtension = globalThis["ohif-extension-default"];
+            const defaultModule = defaultExtension
+              ?.getDataSourcesModule?.()
+              ?.find((module) => module.name === "dicomweb");
+
+            if (!defaultModule) {
+              throw new Error("The default OHIF dicomweb data source is unavailable.");
+            }
+
```
[the remark this answers is not in the document]
Comment (3clyp50) [reply viewer/assets/radsysx-ohif-extension.js:385]: Fixed in `viewer/assets/radsysx-ohif-extension.js`. The wrapper now resolves the default OHIF dicomweb module through `extensionManager.getModuleEntry("@ohif/extension-default.dataSourcesModule.dicomweb")` instead of depending on a `globalThis["ohif-extension-default"]` global.

File: deploy/clinical-stack/render-orthanc-config.sh (hunk 3 above)
[the remark this answers is not in the document]
Comment (3clyp50) [reply deploy/clinical-stack/render-orthanc-config.sh:12]: Verified this one before changing the helper. The proposed single-pass escape breaks credentials containing backslashes or quotes during the final sed replacement. I kept the two-stage escape and revalidated it against special-character samples instead of applying the simplification.

File: viewer/assets/radsysx-bootstrap.js [hunk 5]
```diff
@@ -3,63 +3,129 @@
...
-    fail("The OHIF viewer requires a governed launch session.");
-    return;
-  }
+  window.__RADSYSX_BOOTSTRAP_PROMISE__ = bootstrap();
 
   try {
+    await window.__RADSYSX_BOOTSTRAP_PROMISE__;
+  } catch (error) {
+    fail(error instanceof Error ? error.message : "Unable to bootstrap the viewer.");
+  }
```
[the remark this answers is not in the document]
Comment (3clyp50) [reply viewer/assets/radsysx-bootstrap.js:23]: Fixed in `viewer/assets/radsysx-bootstrap.js`. The outer bootstrap failure handler now clears the stored launch token on every terminal failure path, not only in the resolve-failure branch.

File: viewer/assets/radsysx-bootstrap.js [hunk 6]
```diff
@@ -3,63 +3,129 @@
...
+        clearStoredLaunchToken();
+        throw error;
+      }
+    }
+
     window.__RADSYSX_LAUNCH__ = resolved;
     window.__RADSYSX_CLEAN_VIEWER_URL__ = function cleanViewerUrl() {
-      history.replaceState(null, "", window.location.pathname);
+      history.replaceState(history.state, "", window.location.pathname);
     };
```
[the remark this answers is not in the document]
Comment (3clyp50) [reply viewer/assets/radsysx-bootstrap.js:75]: Fixed in `viewer/assets/radsysx-bootstrap.js`. `__RADSYSX_CLEAN_VIEWER_URL__` now reuses `stripSensitiveQuery()` so it removes only the governed launch/UID keys and preserves benign viewer query params.

File: viewer/assets/radsysx-ohif-extension.js [hunk 7]
```diff
@@ -436,4 +487,37 @@
...
+    if (hasSeriesFilter) {
+      return args;
+    }
+
+    return {
+      ...(args ?? {}),
+      filters: {
+        ...filters,
+        seriesInstanceUIDs,
+      },
```
[the remark this answers is not in the document]
Comment (3clyp50) [reply viewer/assets/radsysx-ohif-extension.js:522]: Fixed in `viewer/assets/radsysx-ohif-extension.js`. The injected filter now uses the singular `seriesInstanceUID` key with the existing UID array payload, which matches the OHIF dicomweb datasource expectation.

File: viewer/scripts/build-ohif-dist.mjs [hunk 8]
```diff
@@ -49,20 +49,18 @@ function copyWorkspaceAsset(relativeParts, outputName) {
 function writeAppConfig() {
   const config = `/** @type {AppTypes.Config} */
 (function radsysxAppConfig() {
-  const resolved = window.__RADSYSX_LAUNCH__;
-  const runtime = resolved?.viewerRuntime ?? {};
   window.config = {
     name: "config/radsysx-clinical.js",
-    routerBasename: runtime.viewerBasePath ?? "/viewer",
+    routerBasename: "/viewer",
     extensions: [
```
[the remark this answers is not in the document]
Comment (3clyp50) [reply viewer/scripts/build-ohif-dist.mjs]: Fixed across `viewer/scripts/build-ohif-dist.mjs`, `viewer/assets/radsysx-bootstrap.js`, and `viewer/assets/radsysx-ohif-extension.js`. The generated OHIF config no longer bakes `/viewer` or `/dicom-web`; router base path and DICOMweb roots are now hydrated from the resolved backend `viewerRuntime` at runtime.

**Comment by 3clyp50** [comment]
Commit 0707a519028f14e99833571887b35b07f322b1f7 closes the remaining PR review feedback around the RadSysX clinical OHIF runtime.

The main focus is the viewer/bootstrap/runtime wiring:
- safer governed launch-token handling
- removal of fragile OHIF global lookups
- correct launch-selected series filtering
- runtime-derived viewer and DICOMweb paths instead of baked defaults

It also documents why the Orthanc config helper intentionally stays POSIX `sh` and keeps its two-stage escaping logic.

## What Changed

### Viewer bootstrap hardening

Updated `viewer/assets/radsysx-bootstrap.js` to:

- clear the stored launch token on every terminal bootstrap failure
- fall back to the original `launch` token if `sessionStorage` is unavailable in the current tab
- fail safely instead of losing the governed launch during login handoff when storage cannot persist it
- preserve benign query params by reusing `stripSensitiveQuery()` for post-mount URL cleanup
- derive the viewer base path dynamically and use it for login redirect and runtime config hydration
- publish the resolved `viewerRuntime` onto the bootstrap/runtime seam for downstream OHIF config use

### OHIF datasource reliability fixes

Updated `viewer/assets/radsysx-ohif-extension.js` to:

- remove the fragile dependency on `globalThis["ohif-extension-default"]`
- resolve the default OHIF dicomweb datasource through `extensionManager.getModuleEntry(...)`
- apply backend-emitted DICOMweb roots to the wrapped datasource before the default OHIF datasource initializes
- inject the launch-selected series filter under the singular `seriesInstanceUID` key expected by OHIF

### Generated OHIF config / path handling

Updated `viewer/scripts/build-ohif-dist.mjs` to:

- stop baking `/viewer` into `routerBasename`
- stop baking `/dicom-web` into the generated datasource config
- make the generated `index.html` derive its public/base path dynamically at runtime
- switch injected viewer asset paths to runtime-relative resolution under the computed base path

This keeps the viewer aligned with backend-emitted `viewerRuntime` values instead of assuming the default local deployment mount.

### Orthanc helper guidance

Updated `deploy/clinical-stack/render-orthanc-config.sh` to document that:

- the helper intentionally remains POSIX `sh`
- the two-stage escaping is deliberate because it must survive both JSON string encoding and the final `sed` replacement pass

This was added after revalidating the helper with credentials containing backslashes, quotes, `&`, and `|`.

## Review Follow-Up

In addition to the code changes, all remaining GitHub PR review threads for this branch were addressed directly with `gh`:
- fixed items received a short remediation reply
- non-applied items received a verification/explanation reply
- all 9 review threads were resolved

## Validation

Passed:
- `cmd.exe /c npm.cmd run type-check --workspace viewer`
- `cmd.exe /c npm.cmd run build --workspace viewer`
- Orthanc escaping regression check with special-character credentials

Not rerun:
- backend test suites, because this change set only touched viewer/runtime assets and a shell-script comment

**Comment by 3clyp50** [comment]
Now moving my dev stack to Linux, so no more WSL noise or node run through cmd.
